File: Demo/fsadf/fsadf/ViewController.m
//
//  ViewController.m
//  fsadf
//
//  Created by JustinLau on 2019/6/25.
//  Copyright © 2019 JustinLau. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@property (weak, nonatomic) IBOutlet UIImageView *imageView;
@property (weak, nonatomic) IBOutlet UIView *whiteVewi;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

- (void)touchesBegan:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event {
    UIImage *image = [UIImage imageNamed:@"rect"];
    CGFloat imageWH = image.size.width * 0.5;
    image = [image resizableImageWithCapInsets:UIEdgeInsetsMake(imageWH-1, 3, imageWH -1, imageWH -1) resizingMode:UIImageResizingModeStretch];
    _imageView.image = image;

    _whiteVewi.layer.backgroundColor = [UIColor colorWithRed:255/255.0 green:255/255.0 blue:255/255.0 alpha:1.0].CGColor;
    _whiteVewi.layer.shadowColor = [UIColor yellowColor].CGColor;
    _whiteVewi.layer.shadowOffset = CGSizeMake(0, -3);
    _whiteVewi.layer.shadowOpacity = 1;
    _whiteVewi.layer.shadowRadius = 3.0;
}


@end

File: iOS/SwiftUI/AppleDoc/AutoLayoutTest/AutoLayoutTest/ViewController.swift
//
//  ViewController.swift
//  AutoLayoutTest
//
//  Created by 刘国强 on 2021/5/29.
//

import UIKit

class ViewController: UIViewController {
    
    var redView: UIView!

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        
        let width = self.view.frame.size.width * 0.6;
        let height = self.view.frame.size.height * 0.6;
        var left = self.view.frame.size.width - width;
        left *= 0.5;
        var top = self.view.frame.size.height - height;
        top *= 0.5;
        let subView = UIView()
        subView.frame = CGRect(x: left, y: top, width: width, height: height)
        subView.autoresizingMask = [.flexibleHeight, .flexibleWidth]
        subView.backgroundColor = .red
        self.view.addSubview(subView)
        
        self.redView = subView
        
        
        self.view.safeAreaLayoutGuide
    }

    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
        print(event)
    }

}

